xonsh/main.py

```diff
@@ -253,6 +254,33 @@ def premain(argv=None):
 
 
 def main(argv=None):
+    try:
+        return main_xonsh(argv)
+    except Exception:
+        traceback.print_exc()
```

Comment:
If you import it at the top, you should still have a reference to it.  But I think I see what you are saying, that xonsh builtins might not be around anymore and so `xt.print_exception()` could itself fail.  That is fair.